repairnator/src/main/java/fr/inria/spirals/repairnator/process/inspectors/ProjectInspector4Bears.java

```diff
@@ -24,44 +29,86 @@
+                    AbstractStep gatherTestInformation = new GatherTestInformation(this, new BuildShouldFail());
+
+                    cloneRepo.setNextStep(buildRepo).setNextStep(testProject).setNextStep(this.testInformations)
+                            .setNextStep(checkoutPreviousBuild).setNextStep(buildRepoForPreviousBuild)
+                            .setNextStep(testProjectForPreviousBuild).setNextStep(gatherTestInformation);
+
+                    lastStep = gatherTestInformation;
+                } else {
+                    AbstractStep gatherTestInformation = new GatherTestInformation(this, new BuildShouldPass());
+                    AbstractStep checkoutSourceCodeForPreviousBuild = new CheckoutSourceCodeForPreviousBuild(this);
+                    AbstractStep buildRepoForPreviousBuild2 = new BuildProject(this);
+                    AbstractStep testProjectForPreviousBuild2 = new TestProject(this);
+                    AbstractStep gatherTestInformation2 = new GatherTestInformation(this, new BuildShouldFail());
+
+                    cloneRepo.setNextStep(buildRepo).setNextStep(testProject).setNextStep(this.testInformations)
+                            .setNextStep(checkoutPreviousBuild).setNextStep(buildRepoForPreviousBuild)
+                            .setNextStep(testProjectForPreviousBuild).setNextStep(gatherTestInformation)
+                            .setNextStep(checkoutSourceCodeForPreviousBuild).setNextStep(buildRepoForPreviousBuild2)
+                            .setNextStep(testProjectForPreviousBuild2).setNextStep(gatherTestInformation2);
+
+                    lastStep = gatherTestInformation2;
+                }
 
-        firstStep = cloneRepo;
-        firstStep.setDataSerializer(this.serializers);
+                if (this.getPushMode()) {
+                    PushIncriminatedBuild pushIncriminatedBuild = new PushIncriminatedBuild(this);
+                    pushIncriminatedBuild.setRemoteRepoUrl(PushIncriminatedBuild.REMOTE_REPO_BEAR);
+                    this.setPushBuild(pushIncriminatedBuild);
+                    lastStep.setNextStep(pushIncriminatedBuild);
+                }
 
-        firstStep.setState(ProjectState.INIT);
+                firstStep = cloneRepo;
+                firstStep.setState(ProjectState.INIT);
+                firstStep.setDataSerializer(this.serializers);
 
-        try {
-            firstStep.execute();
-        } catch (Exception e) {
-            this.addStepError("Unknown", e.getMessage());
-            this.logger.debug("Exception catch while executing steps: ", e);
+                try {
+                    firstStep.execute();
+                } catch (Exception e) {
+                    this.addStepError("Unknown", e.getMessage());
+                    this.logger.debug("Exception catch while executing steps: ", e);
+                }
+            } else {
+                this.setState(ProjectState.PREVIOUSVERSIONISNOTINTERESTING);
```

Comment:
Ok, I'll forget the tests for now, but I did a test anyway to this issue pointed out for you, and actually I wouldn't have a NullPointerException because no step was created and then no serialization was made. This is bad because this way I wouldn't have this passing build in my spreadsheet with the `PREVIOUSVERSIONISNOTINTERESTING` status. I fixed that.